### ISSUE TITLE
Implementation of Stateful Page Transitions for Screen Reader/Keyboard Users

### ISSUE THREAD

**Issue by jscholes** [benetech/MathShare#821]
In issue #742, the lack of accessible feedback on page changes was addressed.  This greatly improves the usability and accessibility of MathShare for screen reader users.

This ticket specifically relates to stateful transitions, in cases where a more appropriate keyboard focus target than the main page heading can be determined based on previous user activity.  This includes:

* When a user returns to the "All Problem Sets" page from within a specific problem set, keyboard focus should land on the link representing that set.
* When a user returns to a list of problems, keyboard focus should land on the link representing the problem most recently accessed.  Ideally, this would span across browser sessions, so that a user can simply continue from where they left off.

Aha! Link: https://diagramlabs.aha.io/features/B-103

**Comment by rupeshparab** [comment]
I have added focus on moving back to a visited page, this focus is handled by storing the current page link in Redux (This data is cleared on reload)
In some cases, the link changes on opening (in case of solve links), for such cases we are storing the previous link as well as the link before redirection and we use the link present in DOM to focus element.

> Ideally, this would span across browser sessions, so that a user can simply continue from where they left off.

If needed we can persist this data in backend for logged in users, but we won't be able persist it locally for users who are not logged in as in some cases the Computers are shared between multiple students.

**Comment by jscholes** [comment]
Implemented in https://github.com/benetech/MathShare/commit/8bfb4d8ef788d65a222dc1186f33d5cf7c6a2096.  @rupeshparab - great job on this one.

**Comment by jscholes** [comment]
@rupeshparab - I'm reopening this due to a small issue which hopefully we can swiftly address.

If the page that the user returns to contains multiple links to the same destination, the first of these will always receive focus.  For example:

1. From the Overview page, activate the "Open Source" link.
2. Press the Back button.

Focus should land on "Open Source", but instead it lands on "Partners".

**Comment by rupeshparab** [comment]
@jscholes 
I have made some tweaks to capture the xpath of the anchor to focus the correct element

It seems to be working with Open Source/Partners and the 2 sign in links on the dashboard page now

**Comment by jscholes** [comment]
Fixed in https://github.com/benetech/MathShare/commit/9d5e69f58909f0ed8f4cf035a82a5aabf937adbb.
